docs/source/mpi4py.util.pool.rst

```diff
@@ -0,0 +1,75 @@
+mpi4py.util.pool
+----------------
+
+.. module:: mpi4py.util.pool
+   :synopsis: :mod:`multiprocessing.pool` interface via :mod:`mpi4py.futures`.
+
+.. versionadded:: 4.0.0
+
+.. seealso::
+
+   This module intends to be a drop-in replacement for the
+   :mod:`multiprocessing.pool` interface from the Python standard library.
+   The :class:`~mpi4py.util.pool.Pool` class exposed here is implemented as a
+   thin wrapper around :class:`~mpi4py.futures.MPIPoolExecutor`.
+
+.. note::
+
+   The :mod:`mpi4py.futures` package offers a higher level interface
+   for asynchronously pushing tasks to MPI worker process, allowing
+   for a clear separation between submitting tasks and waiting for the
+   results.
+
+
+.. autoclass:: mpi4py.util.pool.Pool
+
+   .. automethod:: __init__
+
+   .. automethod:: apply
+
+   .. automethod:: apply_async
+
+   .. automethod:: map
+
+   .. automethod:: imap
+
+   .. automethod:: imap_unordered
+
+   .. automethod:: map_async
+
+   .. automethod:: starmap
+
+   .. automethod:: istarmap
+
+   .. automethod:: istarmap_unordered
+
+   .. automethod:: starmap_async
+
+   .. automethod:: close
+
+   .. automethod:: terminate
+
+   .. automethod:: join
+
+
+.. autoclass:: mpi4py.util.pool.ThreadPool
+   :show-inheritance:
+
+
+.. autoclass:: mpi4py.util.pool.AsyncResult
+
+   .. automethod:: ready
+
+   .. automethod:: successful
+
+   .. automethod:: wait
+
+   .. automethod:: get
+
+
+.. autoclass:: mpi4py.util.pool.ApplyResult
+   :show-inheritance:
+
+
+.. autoclass:: mpi4py.util.pool.MapResult
+   :show-inheritance:
```

Comment:
One time I write comprehensive docs and you ask me to get rid of things?? Now seriously, Python docs are not necessarily prime examples of quality. They document `AsyncResult` as a class, but it is actually an alias, because they use to have a documentation bug. I prefer to keep them documented. Remember mpi4py provide typing stubs. Smart IDEs may show that `apply_async()`/`map_async()` return `ApplyResult`/`MapResult`, so it is good to have an entry in the docs for them.

Reply:
Good point on typing. I haven't thought about that.